Minor edit to CONTRIBUTING.md to list checks (#44)
epiverse-trace/.github: CONTRIBUTING.md

- Make your changes, commit to git, and then create a PR by running `usethis::pr_push()`, and following the prompts in your browser. The title of your PR should briefly describe the change. The body of your PR should contain Fixes #issue-number.
- For user-facing changes, add a bullet to the top of `NEWS.md` (i.e. just below the first header). Follow the style described in https://style.tidyverse.org/news.html.

### What happens after submitting a pull request?

PRs are reviewed by the team before they are merged. The review process begins after continuous integration automated checks, which can take some time. We do not expect contributors to familiarise themselves with all the automated checks in our GitHub actions, and are happy to support on this once a PR is made. 

> However, for those who are interested in running local checks to accelerate the PR process, here is a summary of the main checks that are automatically performed and how they can be confirmed:

Reviewer: ```suggestion
If possible, we recommend you run the automated checks on your local computer first for easier troubleshooting and a reduced cloud computing resource usage. Here is a summary of the main checks that are automatically performed and how they can be confirmed:
```

- Make your changes, commit to git, and then create a PR by running `usethis::pr_push()`, and following the prompts in your browser. The title of your PR should briefly describe the change. The body of your PR should contain Fixes #issue-number.
- For user-facing changes, add a bullet to the top of `NEWS.md` (i.e. just below the first header). Follow the style described in https://style.tidyverse.org/news.html.

### What happens after submitting a pull request?

> PRs are reviewed by the team before they are merged. The review process begins after continuous integration checks, which can take some time. We do not expect contributors to familiarise themselves with all the automated checks in our GitHub actions, and are happy to support on this once a PR is made. 

Reviewer: ```suggestion
PRs are reviewed by the team before they are merged. The review process begins after continuous integration automated checks, which can take some time. We do not expect contributors to familiarise themselves with all the automated checks in our GitHub actions, and are happy to support on this once a PR is made. 
```

> However, for those who are interested in running local checks to accelerate the PR process, here is a summary of the main checks that are automatically performed and how they can be confirmed:

Reviewer 2: So, I understand this to mean that the local checks are optional but I wouldn't suggest that. I would rather suggest that they make sure the local checks are passing and we can help them with the CI check failures. If users are struggling with the local checks, they can always post their log in the issue thread for help. If we don't enforce that, it will become quite a burden (especially in phase 2, where there will be only one RSE) to always fix CI failures that could have been prevented with local checks, and also considering that for large packages, these take some time.

@Bisaloo and @chartgerink thoughts?

Reviewer: What about "we recommend you run local checks" as a middle ground between making it sound like unnecessary and making it mandatory?

> If we don't enforce that, it will become quite a burden (especially in phase 2, where there will be only one RSE) to always fix CI failures that could have been prevented with local checks

I expect contributors who are able to fix local test failures to be able to fix CI test failures as well. As such, I think the maintainer work is more or less the same in both cases.

- Load the package for Epiverse-customised versions of the checks: `remotes::install_github("epiverse-trace/etdev")`.

- Run `lintr::lint_package()` to check for programmatic and stylistic errors in code.

- Run `devtools::document()` to regenerate the .Rd files if changes were made to documentation.

- Run `devtools::check()` to make sure the package passes R CMD check (see previous section). You may prefer to run individual checks such as `spelling::spell_check_test()` to check spelling and `testthat::test_package()` to check that package tests are still passing before running this more comprehensive check.

> 

Reviewer 2: ```suggestion
```

- For user-facing changes, add a bullet to the top of `NEWS.md` (i.e. just below the first header). Follow the style described in https://style.tidyverse.org/news.html.

### What happens after submitting a pull request?

PRs are reviewed by the team before they are merged. The review process begins after continuous integration checks, which can take some time. We do not expect contributors to familiarise themselves with all the automated checks in our GitHub actions, and are happy to support on this once a PR is made. 

However, for those who are interested in running local checks to accelerate the PR process, here is a summary of the main checks that are automatically performed and how they can be confirmed:

- Load the package for Epiverse-customised versions of the checks: `remotes::install_github("epiverse-trace/etdev")`.

- Run `testthat::test_package()` to check that package tests are still passing.

- Run `lintr::lint_package()` to check for programmatic and stylistic errors in code.

- Run `spelling::spell_check_test()` to check spelling and update word list if necessary.

> - Run `devtools::check()` to check overall package structure.

Reviewer 2: This does the spell checking and test coverage check so maybe we don't have to list them explicitly? Maybe you can record to say that `devtools::check()` does a range of checks but if the user wants control over it, they can run the individual checks before running `devtools::check()`.

One additional thing to run is `devtools::document()`. If changes were made to documentation, that function needs to be run to regenerate the `.Rd` files.

Author: Thanks, good suggestions.
